src/java.base/share/native/libjava/jni_util.c

```diff
@@ -654,6 +655,30 @@ static jboolean jnuEncodingSupported(JNIEnv *env) {
     return isJNUEncodingSupported;
 }
 
+/* Tests true if encoding by the name of "UTF-8" is supported by java.nio */
+static jboolean utf8EncodingSupported(JNIEnv *env) {
+    if (utf8Encoding == NULL) { // Cache jstring "UTF-8"
+        jstring utf8Str = (*env)->NewStringUTF(env, "UTF-8");
+        if (utf8Str == NULL) {
+            return JNI_FALSE;
+        }
+        utf8Encoding = (jstring)(*env)->NewGlobalRef(env, utf8Str);
+        (*env)->DeleteLocalRef(env, utf8Str);
+    }
+    static jboolean isUTF8EncodingSupported = JNI_FALSE;
+    if (isUTF8EncodingSupported == JNI_TRUE) {
+        return JNI_TRUE;
+    }
+    jboolean sawException;
+    isUTF8EncodingSupported = (jboolean) JNU_CallStaticMethodByName (
+            env, &sawException,
+            "java/nio/charset/Charset",
+            "isSupported",
+            "(Ljava/lang/String;)Z",
+            utf8Encoding).z;
+    return isUTF8EncodingSupported;
+}
```

Comment:
It would be moot if we choose not to modify the JVM_ interface to standard UTF-8, but this function is not necessary. UTF-8 encoding is guaranteed in every Java implementation.